tiledb/cc/array.cc

```diff
@@ -30,45 +59,46 @@ void init_array(py::module &m) {
 
       // TODO capsule Array(const Context& ctx, tiledb_array_t* carray,
       // tiledb_config_t* config)
-      .def("is_open", &Array::is_open)
-      .def("uri", &Array::uri)
-      .def("schema", &Array::schema)
-      //.def("ptr", [](Array& arr){ return py::capsule(arr.ptr()); } )
-      .def("open", (void(Array::*)(tiledb_query_type_t)) & Array::open)
-      .def("reopen", &Array::reopen)
-      .def("set_open_timestamp_start", &Array::set_open_timestamp_start)
-      .def("set_open_timestamp_end", &Array::set_open_timestamp_end)
-      .def_property_readonly("open_timestamp_start",
+      .def("_is_open", &Array::is_open)
+      .def("_uri", &Array::uri)
+      .def("_schema", &Array::schema)
+      .def("_query_type", &Array::query_type)
+      .def("__capsule__",
+           [](Array &self) { return py::capsule(self.ptr().get(), "array"); })
+      .def("_open", (void(Array::*)(tiledb_query_type_t)) & Array::open)
+      .def("_reopen", &Array::reopen)
+      .def("_set_open_timestamp_start", &Array::set_open_timestamp_start)
+      .def("_set_open_timestamp_end", &Array::set_open_timestamp_end)
+      .def_property_readonly("_open_timestamp_start",
                              &Array::open_timestamp_start)
-      .def_property_readonly("open_timestamp_end", &Array::open_timestamp_end)
-      .def("set_config", &Array::set_config)
-      .def("config", &Array::config)
-      .def("close", &Array::close)
-      .def("consolidate",
+      .def_property_readonly("_open_timestamp_end", &Array::open_timestamp_end)
+      .def("_set_config", &Array::set_config)
+      .def("_config", &Array::config)
+      .def("_close", &Array::close)
+      .def("_get_enumeration",
+           [](Array &self, const Context &ctx, const std::string &attr_name) {
+             return ArrayExperimental::get_enumeration(ctx, self, attr_name);
+           })
+      .def("_consolidate",
            [](Array &self, const Context &ctx, Config *config) {
```

Comment:
The `consolidate` functions should accept a URI instead of an array object.

Reply:
IIUC you mean this signature instead?
https://github.com/TileDB-Inc/TileDB/blob/034ed809077f6b804583438387fe248aabca230a/tiledb/sm/cpp_api/array.h#L588-L594

Reply:
Yes.

Reply:
Good catch, @teo-tsirpanis! There was no point in accepting an array object as an argument since the only thing we're using it for is the uri.